server/src/main/java/keywhiz/commands/BackfillOwnershipCommand.java

```diff
@@ -0,0 +1,165 @@
+package keywhiz.commands;
+
+import static keywhiz.jooq.tables.Accessgrants.ACCESSGRANTS;
+import static keywhiz.jooq.tables.Secrets.SECRETS;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import io.dropwizard.cli.ConfiguredCommand;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.setup.Bootstrap;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import keywhiz.Environments;
+import keywhiz.KeywhizConfig;
+import keywhiz.inject.InjectorFactory;
+import keywhiz.service.config.Readonly;
+import net.sourceforge.argparse4j.inf.Namespace;
+import net.sourceforge.argparse4j.inf.Subparser;
+import org.jooq.Cursor;
+import org.jooq.DSLContext;
+import org.jooq.Record1;
+
+public class BackfillOwnershipCommand extends ConfiguredCommand<KeywhizConfig> {
+  private static final class Args {
+    private static final String BATCH_SIZE = "batchSize";
+    private static final String DELAY = "delay";
+  }
+
+  public BackfillOwnershipCommand() {
+    super("backfill-ownership", "Backfills secret ownership");
+  }
+
+  @Override public void configure(Subparser subparser) {
+    subparser.addArgument("--batchSize")
+        .dest(Args.BATCH_SIZE)
+        .type(Integer.class)
+        .setDefault(1)
+        .help("Number of records to process in one batch");
+
+    subparser.addArgument("--delay")
+        .dest(Args.DELAY)
+        .type(String.class)
+        .setDefault("PT0S")
+        .help("Delay between batches in ISO 8601 duration format e.g. PT0.1s");
+  }
+
+  @Override protected void run(Bootstrap<KeywhizConfig> bootstrap, Namespace namespace,
+      KeywhizConfig config) throws Exception {
+
+    ManagedDataSource dataSource = config.getDataSourceFactory().build(new MetricRegistry(), "backfill-ownership-datasource");
+    Injector injector = InjectorFactory.createInjector(
+        config,
+        Environments.fromBootstrap(bootstrap),
+        dataSource);
+
+    DSLContext jooq = injector.getInstance(Key.get(DSLContext.class, Readonly.class));
+    int batchSize = getBatchSize(namespace);
+    Duration delay = getDelay(namespace);
+
+    new Backfill(jooq, batchSize, delay).run();
+  }
+
+  private static int getBatchSize(Namespace namespace) {
+    Integer batchSize = namespace.getInt(Args.BATCH_SIZE);
+    return batchSize == null
+        ? 1
+        : batchSize;
+  }
```

Comment:
In real-world use this should be unnecessary since we've supplied a default for the argument. However, this also requires us to fake out the parser's behavior during testing, which is annoying. So, I decided to make these methods a bit more robust against missing values than they otherwise would be.

Reply:
Sounds good. Also minor nit: since this particular ternary operation is very short and simple, I think we could get away with putting everything on one line.